### PR TITLE
Implement has-children

### PR DESCRIPTION
**Pull request by mehmetscoskun**

### CODE REVIEW

File: test/specs/parsing/functions/functions.node.tests.ts [hunk 1]
```diff
@@ -332,4 +332,26 @@ return $node/root() = $element`,
 			);
 		});
 	});
+
+	describe('has-children()', () => {
+		it.only('yes, it has children', () => {
```
Comment (DrRataplan) [inline test/specs/parsing/functions/functions.node.tests.ts]: Could you remove the 'only' clauses for the tests?

Comment (DrRataplan) [reply test/specs/parsing/functions/functions.node.tests.ts]: And also, I'm sure you can think of better test names

File: src/expressions/functions/builtInFunctions.node.ts [hunk 2]
```diff
@@ -19,7 +21,7 @@ import FunctionDefinitionType from './FunctionDefinitionType';
 
 const fnString = builtinStringFunctions.functions.string;
 
-function contextItemAsFirstArgument(fn, dynamicContext, executionParameters, _staticContext) {
+function contextItemAsFirstArgument(fn: FunctionDefinitionType, dynamicContext, executionParameters, _staticContext) {
```
Comment (DrRataplan) [inline src/expressions/functions/builtInFunctions.node.ts]: Could you run Prettier over this file?

File: src/expressions/functions/builtInFunctions.node.ts [hunk 3]
```diff
@@ -86,6 +93,22 @@ const fnName: FunctionDefinitionType = function(
...
+const fnHasChildren: FunctionDefinitionType = function(
+	_dynamicContext,
+	_executionParameters,
+	_staticContext,
+	nodeSequence: ISequence
+) {
+	return zipSingleton([nodeSequence], ([nodeValue]: (Value | null)[]) => {
+		const node: ConcreteNode = nodeValue ? nodeValue.value : null;
+
+		if (node && node.firstChild) {
```
Comment (JosVerburg) [inline src/expressions/functions/builtInFunctions.node.ts]: I believe we should use the `executionParameters.domFacade` to retrieve the child node.
Note: `executionParameters` is now `_executionParameters`

**Pull request merged**